File: src/test/java/dev/sanket/lengthoflastword/SolutionTest.java
package dev.sanket.lengthoflastword;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {

    private final Solution solution = new Solution();

    @ParameterizedTest
    @CsvSource(value = {"Hello World:5", "   fly me   to   the moon  :4", "hello:5"}, delimiter = ':')
    public void shouldPass(String input, int expected) {
        assertThat(solution.lengthOfLastWord(input)).isEqualTo(expected);
    }
}

File: src/test/kotlin/dev/sanket/sqrt/SolutionTest.kt
package dev.sanket.sqrt

import assertk.assertThat
import assertk.assertions.isEqualTo
import dev.sanket.dev.sanket.sqrt.Solution
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class SolutionTest {

    private val solution = Solution()

    @ParameterizedTest
    @CsvSource(value = ["4:2", "8:2", "1:1", "6:2"], delimiter = ':')
    fun shouldPass(input: String, output: String) {
        val sqrt = solution.mySqrt(input.toInt())
        assertThat(sqrt).isEqualTo(output.toInt())
    }
}